src/main/java/ch/securify/decompiler/DecompilerFallback.java

```diff
@@ -100,9 +100,9 @@
 			for (int offset = 0; offset < rawInstructions.length; offset = ArrayUtil.nextNonNullIndex(offset, rawInstructions)) {
```

Comment:
I haven't look into it, I can do this later, but can you explain what the connection with this PR is?

Reply:
In this PR you are fixing a piece of code that we don't need to fix, because we can simply remove it.

Reply:
OK I'll double-check that it's dead code.

Reply:
Well actually it's clear that it's not dead code, given that it does fix some bugs, e.g., with the empty contract and so on, and the other changes are simply introducing functions instead of listing the same opcodes every time. I'll still look into why this is needed.

Reply:
Thanks. Would be good to double-check and then we can remove it. 

Maybe introducing the fix you proposed for this for the Decompiler.java makes sense. (It has the same code.) Possibly fixing some function detection there. But that is part of a bigger open issue.

Reply:
Yes, I've changed the code in Decompiler.java.

Reply:
I also double-checked, by running tests, and just by inspecting the code by removing useless instructions until I ended up with none, so it looks good to me.